### PR TITLE
vaultwarden: update to 1.21.0, rename from bitwarden_rs

### PR DESCRIPTION
**Pull request by K900**
I tried to make this as non-breaking as possible, but it will still
break things slightly for people expecting certain file names in the
packages themselves.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [x] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by K900** [comment]
Added a release note, it seems appropriate judging by the other ones.

**Comment by K900** [comment]
I'm not sure why ofborg is failing here. Is it because it can't fetch the source?

**Comment by K900** [comment]
I gave up on making ofborg play nice with Cargo.lock, let's just stick to the old way for now.

File: nixos/doc/manual/from_md/release-notes/rl-2111.section.xml
Comment (aanderse) [inline nixos/doc/manual/from_md/release-notes/rl-2111.section.xml:81]: Would the present state of the code would cause this service to fail for existing users because of ownership permissions?

**Review by aanderse — COMMENTED** [review]
I've listed a few things you could do to make this change backwards compatible for existing users, if you were inclined to do so.

File: nixos/modules/services/security/vaultwarden/default.nix
Comment (K900) [reply nixos/modules/services/security/vaultwarden/default.nix:8]: Systemd should sort out the permissions on `/var/lib/bitwarden_rs` automatically, so this shouldn't be a problem unless people go out of their way to modify the users in their own config, which I really hope no one does.

File: nixos/doc/manual/from_md/release-notes/rl-2111.section.xml
Comment (K900) [reply nixos/doc/manual/from_md/release-notes/rl-2111.section.xml:81]: Nope, systemd will automatically change the ownership of `StateDirectory` to the user that owns the service, so we're good here.

File: nixos/modules/services/security/vaultwarden/default.nix
Comment (K900) [reply nixos/modules/services/security/vaultwarden/default.nix:141]: That will still require unstable users to migrate manually, right?

File: nixos/modules/services/security/vaultwarden/default.nix
Comment (K900) [reply nixos/modules/services/security/vaultwarden/default.nix:119]: Thanks, added!

File: nixos/modules/services/security/vaultwarden/default.nix
Comment (aanderse) [reply nixos/modules/services/security/vaultwarden/default.nix:8]: Gosh you're right! I was mistaken.

So what is the incompatibility this PR introduces?

File: nixos/modules/services/security/vaultwarden/default.nix
Comment (aanderse) [reply nixos/modules/services/security/vaultwarden/default.nix:141]: Ignore this, see above comment.

File: nixos/doc/manual/from_md/release-notes/rl-2111.section.xml
Comment (aanderse) [reply nixos/doc/manual/from_md/release-notes/rl-2111.section.xml:81]: Ignore this, see above comment.

File: nixos/modules/services/security/vaultwarden/default.nix
Comment (K900) [reply nixos/modules/services/security/vaultwarden/default.nix:8]: People manually modifying the users/units in their configuration will have things explode because the unit is no longer there. In most cases it will explode in a pretty obvious way, so this is probably fine?

Comment (K900) [reply nixos/modules/services/security/vaultwarden/default.nix:8]: Also anything that depends on the paths in the final derivations, like custom units and such.

File: nixos/doc/manual/from_md/release-notes/rl-2111.section.xml
Comment (aanderse) [inline nixos/doc/manual/from_md/release-notes/rl-2111.section.xml:81]: You should list explicitly that the user and service names changed and anything else that that might be relevant. If users are required to take any action you could mention here, if not you could mention that was well.

**Review by aanderse — CHANGES_REQUESTED** [review]
I think the change looks good. Thanks for your work on this.

Could you please just add a little more detail to the release notes and I think we're good to go.

File: nixos/modules/services/security/vaultwarden/default.nix
Comment (Mic92) [reply nixos/modules/services/security/vaultwarden/default.nix:141]: Right. It could move atomically from the old to the new location so.

File: nixos/doc/manual/from_md/release-notes/rl-2111.section.xml
Comment (K900) [reply nixos/doc/manual/from_md/release-notes/rl-2111.section.xml:81]: I've added a more detailed release note, but I think it might be _too_ detailed now.

**Comment by K900** [comment]
Rebased on top of latest master, is there anything else stopping this from getting merged?

File: nixos/doc/manual/from_md/release-notes/rl-2111.section.xml
Comment (aanderse) [reply nixos/doc/manual/from_md/release-notes/rl-2111.section.xml:322]: hahaha probably - but that is much better than too little :+1:

**Review by aanderse — APPROVED** [review]
LGTM but I would prefer if @msteen, or anyone else familiar with this package & module, to approve if possible before we merge.

If we don't hear anything back in a week or so ping me and we'll go ahead and merge.

**Comment by K900** [comment]
Rebased again, and updated to 1.22.1. I squashed all the commits because rebasing individual release notes every time was too obnoxious.

**Pull request merged**
